test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/ODataDynamicPropertyTests.cs

```diff
@@ -0,0 +1,868 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataDynamicPropertyTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace AstoriaUnitTests.TDD.Tests.Client
+{
+    using Microsoft.OData.Client;
+    using Microsoft.OData.Edm;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.IO;
+    using System.Linq;
+
+    [TestClass]
+    public class ODataDynamicPropertyTests
+    {
+        private EdmModel serviceEdmModel;
+        private EdmEnumType genreEnumType;
+        private EdmComplexType addressComplexType;
+        private EdmComplexType nextOfKinComplexType;
+        private EdmEntityType directorEntityType;
+        private EdmEntityType editorEntityType;
+        private EdmEntityType producerEntityType;
+        private EdmEntityType actorEntityType;
+        private EdmEntityType awardEntityType;
+        private string genreEnumTypeName;
+        private string addressComplexTypeName;
+        private string nextOfKinComplexTypeName;
+        private string directorEntityTypeName;
+        private string directorEntitySetName;
+        private string editorEntityTypeName;
+        private string editorEntitySetName;
+        private string producerEntityTypeName;
+        private string producerEntitySetName;
+        private string actorEntityTypeName;
+        private string actorEntitySetName;
+        private string awardEntityTypeName;
+        private string awardEntitySetName;
+
+        private ClientEdmModel clientEdmModel;
+        private DataServiceContext context;
+        private IDictionary<string, string> typeNameToEntitySetMapping;

```

Comment:
Most of these seem to be set in the constructor, maybe need to have them as readonly...?

Reply:
@g2mula They're not in a constructor but rather in a method decorated with `TestInitialize` so C# will not allow them to be `readonly`